composer.json

```diff
@@ -18,11 +18,14 @@
     }
   ],
   "require": {
-    "php": ">= 7.0",
-    "ext-mbstring": "*"
+    "php": ">= 5.6",
```

Comment:
Debian `jessie` (current oldstable) release does not have PHP 7, only [PHP 5.6](https://packages.debian.org/jessie/php5). This release will get security maintenance until [end of April 2020](https://wiki.debian.org/LTS). I'm probably not the only one who has to run applications on this platform.

Reply:
I would rather implement proper type specifications for function returns than to support PHP 5.6.

I will not raise the required version for a bugfix. A version that was running on a specific platform should receive fixes for that. This change though is clearly a new feature. If someone wants new features, they should also be willing to update their system.

Furthermore this library requires php 7 for a few months now and there were no complaints about that.

If there is no interest in updating the system, there are still alternatives. With the dotdeb repo, updates to PHP 7.* is really easy.

Reply:
[30%](https://seld.be/notes/php-versions-stats-2017-2-edition) of users use PHP 5.x. They will not be able to use these features if you set PHP requirement to 7. Code was explicitly designed for pre-7, `zlib.deflate` stream filter is only used when `deflate_init` function is not available. That was the only point where this library needed PHP 7.

Reply:
The link you've posted states this:

> Please consider bumping to PHP 7.0 at the very least when you update your libraries.

I'd rather spend my time and add type safety to the code and let users of current versions benefit from the modifications instead of supporting users with a version that was released over 3 years ago.

Since you're cleaned up the code base so much, adding types to the code base would be my next PR, which would at least require PHP 7.0 (or PHP 7.1 if there are nullable types present).